cpp/src/Slice/Parser.cpp

```diff
@@ -1178,23 +1192,15 @@ Slice::Container::createClassDef(const string& name, int id, const ClassDefPtr&
     }
 
     ClassDefPtr def = make_shared<ClassDef>(shared_from_this(), name, id, base);
-    _unit->addContent(def);
```

Comment:
Whenever you define a class, we always make sure there's a corresponding `Decl`. If there isn't, we create one here.

Previously, we would add the `Def` to the module first, then add the `Decl` 2nd.
With my changes though, this causes metadata warnings to be emitted in a lame order.

I swapped the order, so now we add the `Decl` first, and then add the `Def` afterwards.
This fixes the order, and also... it's just simpler.